### ISSUE TITLE
ncu-ci commit TypeError: Cannot read properties of null (reading '1')

### ISSUE THREAD

**Issue by F3n67u** [nodejs/node-core-utils#639]
## Reproduce

```
$ ncu-ci commit 54550                      
--------------------------------------------------------------------------------
[1/1] Running COMMIT: 54550
--------------------------------------------------------------------------------
✔  Build data downloaded
⠏ Querying API for job/node-test-binary-windows-js-suites/RUN_SUBSET=2,nodes=win10-arm64-COMPILED_BY-vs2019-arm64/13778/TypeError: Cannot read properties of null (reading '1')
    at new JSTestFailure (file:///usr/local/lib/node_modules/node-core-utils/lib/ci/ci_failure_parser.js:78:62)
    at file:///usr/local/lib/node_modules/node-core-utils/lib/ci/ci_failure_parser.js:168:16
    at Array.map (<anonymous>)
    at filter (file:///usr/local/lib/node_modules/node-core-utils/lib/ci/ci_failure_parser.js:167:29)
    at CIFailureParser.parse (file:///usr/local/lib/node_modules/node-core-utils/lib/ci/ci_failure_parser.js:296:22)
    at TestRun.parseConsoleText (file:///usr/local/lib/node_modules/node-core-utils/lib/ci/build-types/job.js:93:26)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Promise.all (index 13)
    at async CommitBuild.getResults (file:///usr/local/lib/node_modules/node-core-utils/lib/ci/build-types/commit_build.js:106:25)
    at async JobCommand.drain (file:///usr/local/lib/node_modules/node-core-utils/bin/ncu-ci.js:352:7)
```

## Expected behavior
No Error happened, could process `job/node-test-commit/54550/` normally

## Actual behavior 
TypeError thrown

## Additional Information

Refs: https://github.com/nodejs/reliability/issues/316

Jenkins job URL: 
https://ci.nodejs.org/job/node-test-commit-osx-arm/6298/
https://ci.nodejs.org/job/node-test-commit/54550/
https://ci.nodejs.org/job/node-test-pull-request/44957/

**Comment by aduh95** [comment]
The CI has been locked in preparation for the security release, I think the error is the expected behavior.

**Comment by richardlau** [comment]
This was occuring at the end of last week before the CI was locked down, see https://github.com/nodejs/reliability/issues/316.

**Comment by F3n67u** [comment]
After some debugging, I found the following additional info, but still don't figure out why the test console text is not behaving as expected.

Jenkins job URL: 
https://ci.nodejs.org/job/node-test-commit-osx-arm/6298/
https://ci.nodejs.org/job/node-test-commit/54550/
https://ci.nodejs.org/job/node-test-pull-request/44957/

Problem Console Text: 
https://ci.nodejs.org/job/node-test-commit-osx-arm/nodes=osx11/6298/consoleText
https://ci.nodejs.org/job/node-test-commit-osx-arm/6298/nodes=osx11/consoleFull

Pr where the job trigger: 
https://github.com/nodejs/node/pull/43549

The console text which cause parse error:

```console
outline = not ok 3 - sync fail todo # TODO
outline =   ---
outline =   duration_ms: 0.000040417
outline =   failureType: 'testCodeFailure'
outline =   error: 'thrown from sync fail todo'
outline =   code: 'ERR_TEST_FAILURE'
outline =   stack: |-
outline =     Object.<anonymous> (/Users/iojs/build/workspace/node-test-commit-osx-arm/nodes/osx11/test/message/test_runner_desctibe_it.js:17:9)
outline =     ItTest.runInAsyncScope (node:async_hooks:203:9)
outline =     ItTest.run (node:internal/test_runner/test:338:15)
outline =     Test.processPendingSubtests (node:internal/test_runner/test:158:27)
outline =     ItTest.postRun (node:internal/test_runner/test:390:19)
outline =     ItTest.run (node:internal/test_runner/test:352:10)
outline =     async Test.processPendingSubtests (node:internal/test_runner/test:158:7)
outline =   ...
```
Problem points: 
1. There is no `severity` attribute. 
2. The weird `outline =` prefix. 

The following is a console text which will not parse fail:
```console
not ok 967 parallel/test-http-client-response-timeout
  ---
  duration_ms: 0.163
  severity: fail
  exitcode: 1
  stack: |-
    Mismatched <anonymous> function calls. Expected exactly 1, actual 0.
        at Proxy.mustCall (/Users/iojs/build/workspace/node-test-commit-osx-arm/nodes/osx11/test/common/index.js:396:10)
        at ClientRequest.<anonymous> (/Users/iojs/build/workspace/node-test-commit-osx-arm/nodes/osx11/test/parallel/test-http-client-response-timeout.js:10:32)
        at ClientRequest.<anonymous> (/Users/iojs/build/workspace/node-test-commit-osx-arm/nodes/osx11/test/common/index.js:438:15)
        at Object.onceWrapper (node:events:652:26)
        at ClientRequest.emit (node:events:537:28)
        at HTTPParser.parserOnIncomingClient [as onIncoming] (node:_http_client:659:27)
        at HTTPParser.parserOnHeadersComplete (node:_http_common:117:17)
        at Socket.socketOnData (node:_http_client:512:22)
        at Socket.emit (node:events:537:28)
  ...
```
from https://ci.nodejs.org/job/node-test-commit-osx-arm/nodes=osx11/6402/consoleText, https://ci.nodejs.org/job/node-test-commit-osx-arm/6402/console

**Comment by F3n67u** [comment]
> Problem points:
> 
> 1. There is no `severity` attribute.
> 2. The weird `outline =` prefix.

@MoLow do you have any idea why?

**Comment by F3n67u** [comment]
pattern length and outline length differs.
pattern length is 512
outline length is 513

the diff is from last part.

pattern is:

```console
pattern = ^\#\ tests\ 54$
pattern = ^\#\ pass\ 23$
pattern = ^\#\ fail\ 17$
pattern = ^\#\ skipped\ 9$
pattern = ^\#\ todo\ 5$
pattern = ^\#\ duration_ms\ .*$
```

outline is:

```console
outline = # tests 54
outline = # pass 23
outline = # fail 17
outline = # cancelled 0
outline = # skipped 9
outline = # todo 5
outline = # duration_ms 4.798118667
```

the pattern has not `cancelled` but the outline has `cancelled`. this is why length differs print and cause the parse error.

**Comment by F3n67u** [comment]
I know what is the problem now: 

1. `test_runner_desctibe_it` has a snapshot: `test_runner_desctibe_it.out`, but the actual output is not match the snapshot
2. the `test_runner_desctibe_it` fail and print a log that says the `pattern` and `outline` is not match
3. the outline is a test output, ncu-ci think it is a real test output and try to parse it and result in parse fail

**Comment by MoLow** [comment]
@F3n67u anything I can do to help? 
I guess we can either change the output of `tools/test.py` in case of a output that contains tap
or we can change the code in this repo to accept the malformed output.
Il try playing around with both options

**Comment by MoLow** [comment]
Oh thanks for fixing it :)

**Comment by F3n67u** [comment]
> @F3n67u anything I can do to help? I guess we can either change the output of `tools/test.py` in case of a output that contains tap or we can change the code in this repo to accept the malformed output. Il try playing around with both options

I figure out how to fix this problem. I made a pr to fix this.

**Comment by richardlau** [comment]
FWIW we may have a similar issue in https://github.com/nodejs/tap2junit when parsing failures from `test_runner_desctibe_it`, https://github.com/nodejs/node/issues/43773

**Comment by MoLow** [comment]
Seems like the issue can be closed: https://github.com/nodejs/reliability/issues/321

**Comment by F3n67u** [comment]
> fix: parse ci failure error #640

no worry. this issue will closed by https://github.com/nodejs/node-core-utils/pull/640

**Comment by F3n67u** [comment]
Update CI reliability job stop working again, review wanted for https://github.com/nodejs/node-core-utils/pull/640. 
https://github.com/nodejs/reliability/runs/7587812485?check_suite_focus=true
